var/spack/repos/builtin/packages/xorgproto/package.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+from spack.package import *
+
+
+class Xorgproto(MesonPackage):
+    """X Window System unified protocol definitions replacing standalone protos"""
+
+    homepage = "https://gitlab.freedesktop.org/xorg/proto/xorgproto"
+    url = "https://gitlab.freedesktop.org/xorg/proto/xorgproto/-/archive/xorgproto-2024.1/xorgproto-xorgproto-2024.1.tar.bz2"
```

Comment:
XorgPackage has mirrors, e.g.
```suggestion
    xorg_mirror_path = "proto/xorgproto-2024.1.tar.gz"
```
Mirrors only publish gz (for all version) and bz2 (older version) or xz (newer versions), so sticking to gz is most portable.